pkg/apis/autoscaling/v1alpha1/pa_lifecycle.go

```diff
@@ -82,11 +83,16 @@ func (pa *PodAutoscaler) annotationFloat64(key string) (float64, bool) {
 // `(min, max int32)`. The value of 0 for any of min or max means the bound is
 // not set.
 // Note: min will be ignored if the PA is not reachable
-func (pa *PodAutoscaler) ScaleBounds() (min, max int32) {
+func (pa *PodAutoscaler) ScaleBounds(asConfig *autoscalerconfig.Config) (min, max int32) {
 	if pa.Spec.Reachability != ReachabilityUnreachable {
 		min, _ = pa.annotationInt32(autoscaling.MinScaleAnnotationKey)
 	}
-	max, _ = pa.annotationInt32(autoscaling.MaxScaleAnnotationKey)
+
+	if paMax, ok := pa.annotationInt32(autoscaling.MaxScaleAnnotationKey); ok {
+		max = paMax
+	} else {
+		max = asConfig.DefaultMaxScale
+	}
 
 	return
```

Comment:
I know we used it here already, but it'd be really nice to avoid the naked return here, especially since the method is a bit more complicated now

Reply:
fixed; only thing to consider is it's slightly more error-prone now (as in, returning the min and max in the wrong order). since it'd documented above it should be fine.